File: mlpack/pca.cpp
#include <mlpack/core.hpp>
#include "pca.hpp"

using namespace mlpack;
using namespace mlpack::pca;

int main(int argc, char** argv)
{
    arma::mat data, transformedData, coeff;
    arma::vec eigVal;
    
    // Load the data.
    data::Load("../data/ingredients.csv", data, true);
    
    // Perform PCA.
    PCA p;
    p.Apply(data, transformedData, eigVal, coeff);
    
    // Show transform data into eigenvector basis.
    Log::cout << transformedData << std::endl;
    
    return 0;
}

File: mlpy/pca.py
import numpy as np
import os
import mlpy

# Due to an error (AttributeError: 'module' object has no attribute 'PCA').
# I have manually added a modified class file.
from mlpyPCA import PCA

# Load the data.
f = open(os.path.dirname(__file__) + '../data/ingredients.data')
data = np.fromfile(f, dtype=np.float64, sep=' ')
data = data.reshape(-1, 4)
f.close()

# Perform PCA.
pca = PCA()
pca.learn(data)
transformedData = pca.transform(data)

# Show transform data into eigenvector basis.
print transformedData